lib/bitpagos/client.rb

```diff
@@ -22,42 +22,67 @@ def get_transaction(transaction_id)
       retrieve_transactions(nil, transaction_id)
     end
 
-    def all_transactions
-      retrieve_transactions
+    # Retrieve all transactions, or specify them with parameters, to retrieve
+    # allow retrieving paginated transactions.
+    #
+    # @param [Hash] Allows to specify transaction type, limit or offset
+    # @option [Symbol] status (:pending, :waiting, :completed, :partially_paid)
+    # @option [Integer] offset
+    # @option [Integer] limit
+    # @return [Hash]
+    def transactions(params = {})
+      if status = params[:status]
+        params[:status] = get_transaction_type_from_symbol(status)
+      end
+      retrieve_transactions(params)
     end
 
     def completed_transactions
-      retrieve_transactions(COMPLETED)
+      retrieve_transactions({status: COMPLETED})
     end
 
     def waiting_transactions
-      retrieve_transactions(WAITING)
+      retrieve_transactions({status: WAITING})
     end
 
     def pending_transactions
-      retrieve_transactions(PENDING)
+      retrieve_transactions({status: PENDING})
     end
 
     def partially_paid_transactions
-      retrieve_transactions(PARTIALLY_PAID)
+      retrieve_transactions({status: PARTIALLY_PAID})
     end
 
     # Returns the total count of transactions in all states.
     #
     # @return [Integer] Total transaction count
     def transaction_count
-      all_transactions["meta"]["total_count"]
+      transactions["meta"]["total_count"]
     end
 
     private
 
+    # Takes a symbol and returns the proper transaction type.
+    #
+    # @param [Symbol] Can be :pending, :waiting, :completed or :partially_paid
+    # @return [String] Returns the corresponding "PE", "WA", "CO" or "PP"
```

Comment:
@mauro-oto I guess this method could return `nil` too. If so, the documentation should say: 

```ruby
 @return [String,nil]
```